scripts/vehicle_node.py

```diff
@@ -0,0 +1,371 @@
+        self.cloud = self.create_terrain_map()
+
+        #Create Joint State Message
+        self.js = JointState()
+
+        #Create Twist variables to store cmd_vel
+        self.twist = Twist()
+        self.joint_twist = Twist()
+
+        #Create variables for calculations
+        self.vehicle_yaw = 0.0
+        self.vehicle_pitch = 0.0
+        self.vehicle_roll = 0.0
+        self.vehicle_x = 0.0
+        self.vehicle_y = 0.0
+        self.vehicle_z = 0.0
+
+        self.joint_yaw = -1.0
+        self.joint_pitch = 0.0
+        self.joint_roll = 0.0
+        self.joint_x = 0.0
+        self.joint_y = 0.0
+        self.joint_z = 0.0
+
+        self.cloud_msg = PointCloud2()
+
+        #Timer to update pose every 50ms
+        rospy.Timer(rospy.Duration(0.05), self.update_position)
+
+        #Timer to publish Joint msg and pose for joint every 10ms (doesn't work yet)
+        #rospy.Timer(rospy.Duration(0.05), self.update_joint_positions)
+
+        #Keeps program running until interrupted
+        rospy.spin()
+
+    def create_terrain_map(self):
+
+        #Get parameters
+        mu = rospy.get_param('/mu')
+        sigma = rospy.get_param('/sigma')
+        sigma_filt = rospy.get_param('/sigma_filt')
+        height = rospy.get_param('/height')
+        width = rospy.get_param('/width')
+        resolution =  rospy.get_param('/resolution')
+
+        #Fill array of size h x w with Gaussian Noise given the height and width of the point cloud
+        h = int(((height*2) + 1)*(resolution*1000))
+        w = int(((width*2) + 1)*(resolution*1000))
+        grid_map = (h,w)
+        gaussian_array = np.random.normal(mu, sigma, grid_map)
+
+        #Filter the array to smoothen the variation of the noise
+        gaussian_array = gaussian_filter(gaussian_array, sigma_filt)
+
+        #Create an list and fill the list with x, y values and z the values will be the filtered gaussian noise
+        # TODO perform these calculations without for loops
+        incr_x = 0
+        incr_y = 0
+        cloud = []
+        for x in np.arange(-height, (height+1), resolution):
+            for y in np.arange(-width, (width+1), resolution):
+                innerlist = []
+                innerlist.append(x)
+                innerlist.append(y)
+                innerlist.append(gaussian_array[incr_x][incr_y])
+                cloud.append(innerlist)
+                incr_y+=1
+            incr_x+= 1
+            incr_y = 0
+
+        return cloud
+
+
+    def velocity_cmd_callback(self, data):
+
+        #assign updated command velocities to twist variable
+
+        self.twist = data
+
+
+    def update_position(self, event):
+
+        #Load parameters
+        resolution =  rospy.get_param('/resolution')
+        v_height =  rospy.get_param('/v_height')
+        v_width =  rospy.get_param('/v_width')
+
+        #Local variable to calculate position based on most recent command velocity data
+        position_data = self.twist
```

Comment:
I don't think this line does what you think it does:
https://medium.com/broken-window/many-names-one-memory-address-122f78734cb6